app/models/concerns/audit_extensions.rb

```diff
@@ -235,10 +235,10 @@ def set_taxonomy_for(taxonomy)
   def set_taxonomies_using_associated(key_name)
     ids_arr = []
     if associated.respond_to?(:"#{key_name}_id")
-      ids_arr = [associated.send("#{key_name}_id")].compact.uniq
+      ids_arr = [associated.send("#{key_name}_id")]
     elsif associated.respond_to?(:"#{key_name}_ids")
       ids_arr = associated.send("#{key_name}_ids")
     end
-    send("#{key_name}_ids=", ids_arr)
+    send("#{key_name}_ids=", ids_arr.compact.uniq)
```

Comment:
Sorry, one last nit: Looking at the code again, instead of assigning the value to the variable `ids_arr` multiple times, can we take the variable out?
```
ids_arr =
if associated.respond_to?(:"#{key_name}_id")
  [associated.send("#{key_name}_id")]
elsif associated.respond_to?(:"#{key_name}_ids")
  associated.send("#{key_name}_ids")
end
```

Let me know your thoughts?